### ISSUE TITLE
fig.cap not working for github_document

### ISSUE THREAD

**Issue by ttimbers** [rstudio/rmarkdown#1756]
Setting `fig.caption` in a code chunk is currently not working for the `github_document` output format. Here is a reprex:

````
---
title: "Test figure caption"
author: "Tiffany A. Timbers </br>"
date: "2019/12/30 (updated: `r Sys.Date()`)"
output: 
  github_document:
    toc: true
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(knitr)
```

## Figure

```{r pressure, echo=FALSE, fig.cap="A caption", out.width = '50%'}
plot(pressure)
```

````

If I switch the output of this file to `html_document` or `pdf_document` I do get figure captions.

**Comment by cderv** [comment]
Hi! 

Thanks for the report ! 

I think this is because `knitr` does not recognize `github_markdown` when using `pandoc 2+, which `gfm` for pandoc, as "html like" output. 
https://github.com/rstudio/rmarkdown/blob/b5a6ba3e417622cac4c0a2da64a55953f8c62fd6/R/github_document.R#L40

I think the fix should be in knitr here, to add `gfm` as a know markdown format.
https://github.com/yihui/knitr/blob/33d69c3acc0bb0a4f840e9ab836d0a4ec795c9fa/R/utils.R#L340-L346

**Comment by yihui** [comment]
@cderv Thanks for the tip!

@ttimbers Thanks for the report! It should be fixed in **knitr** now. You can install the development version via

```r
remotes::install_github('yihui/knitr')
```

**Comment by ijlyttle** [comment]
I'm writing this note in response to @yihui's [blog post](https://yihui.org/en/2020/02/rstudio-conf-2020/); this is what I *should* have written here at the time. 

As I *now* understand things, the `knitr::is_html_output()` function is written for Pandoc's benefit, answering the question, "will Pandoc think it is writing an HTML format"? 

I had a slightly different question in mind - how to `knit.print()` an htmlwidget, in this case a [vegawidget](https://vegawidget.github.io/vegawidget/#new-to-vegawidget-0-2). I was rendering a `github_document`, in this case the vegawidget `README.Rmd` file, and things stopped working. 

In vegawidget, I was using `knitr::is_html_output()` to answer the question "will this figure print as html?", which is a different question than "will Pandoc think it is writing an HTML format?". For a `github_document`, the answer is different.

Fortunately, Yihui had already designed an [escape hatch](https://bookdown.org/yihui/rmarkdown-cookbook/latex-html.html), and I was able to [use it](https://github.com/vegawidget/vegawidget/blob/dcc926c0fa94f5b8b2fd7b717d06e483cad71a94/R/print.R#L85).

At this point, I realized that what I *thought* was a bug really wasn't a bug. I implemented the fix, and made a mental note to tell Yihui the next time I saw him...
